pkg/transport/listener.go

```diff
@@ -92,6 +101,76 @@ func (info TLSInfo) Empty() bool {
 	return info.CertFile == "" && info.KeyFile == ""
 }
 
+func SelfCert(path string, hosts []string) (info TLSInfo, err error) {
```

Comment:
it seems like in that case tlsinfo should also be moved to tlsutil-- but that would be somewhat intrusive; maybe just separate tls stuff into a separate file from `listener.go` for now?